src/averageAges.js

```diff
@@ -1,61 +1,60 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const arrayMan = people.filter(
+    ({ sex, died }) => century
+      ? Math.ceil(died / 100) === century && sex === 'm'
+      : sex === 'm');
+
+  const summaryAge = arrayMan.reduce(
+    (next, { died, born }) => next + (died - born), 0);
+
+  return summaryAge / arrayMan.length;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average ave of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find the other who mention
- * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayWomen = withChildren
+    ? people.filter(({ name, sex }) => people.some(
+      ({ mother }) => name === mother) && sex === 'f')
+    : people.filter(({ sex }) => sex === 'f');
+
+  const summaryAge = arrayWomen.reduce(
+    (next, { died, born }) => next + (died - born), 0);
```

Comment:
you have user the same logic for men. it would be a good idea to create a function with this logic